src/backend/joanie/core/api/admin/__init__.py

```diff
@@ -636,6 +641,50 @@ def generate_certificate(self, request, pk=None):  # pylint:disable=unused-argum
 
         return Response(certificate, status=HTTPStatus.CREATED)
 
+    @action(methods=["POST"], detail=True)
+    def cancel(self, request, pk=None):  # pylint:disable=unused-argument
+        """
+        Cancel an order.
+        """
+        order = self.get_object()
+
+        order.flow.cancel()
+
+        return Response(status=HTTPStatus.OK)
+
+    @action(methods=["POST"], detail=True)
+    def refund(self, request, pk=None):  # pylint:disable=unused-argument
+        """
+        Refund an order only if the order is in state 'cancel' and at least 1 installment
+        has been paid in the payment schedule.
+        """
+        order = self.get_object()
+
+        if order.state != enums.ORDER_STATE_CANCELED:
+            return Response(
+                "Cannot refund an order not canceled.",
+                status=HTTPStatus.UNPROCESSABLE_ENTITY,
+            )
+
+        if not has_installment_paid(order):
+            return Response(
+                "Cannot refund an order without paid installments in payment schedule.",
+                status=HTTPStatus.BAD_REQUEST,
+            )
```

Comment:
All this check should be called when applytin `flow.refund()`. IMO it is useless to do it twice. You should do a `try/except` on the `order.flow.refund()` and return the good response based on what is happening